changelog/59.removal.rst

```diff
@@ -0,0 +1,2 @@
+Remove internal usage of legacy ``__multicall__`` recursive hook calling system.
```

Comment:
Unless I'm misunderstanding, this is a public API, not just internal usage, so 

```suggestion
Remove legacy ``__multicall__`` recursive hook calling system.
```

Reply:
@bluetech actually I'm pretty sure this was always an internal system used as a hack, which is why we were also able to remove it.

I think @RonnyPfannschmidt might be able to comment since he has seniority :smirk_cat: